src/main.ts

```diff
@@ -0,0 +1,75 @@
+/***************************************************************
+ * node/context/interpolate.js
+ *
+ * Copyright (C) Lolocodo AB - All rights reserved
+ ***************************************************************/
+
+
+import { get, reduce } from 'lodash';
+
+export const interpolate = (ctx: unknown, params: unknown): unknown => {
+  return deepMap(params, val => interpolateOne(ctx, val));
+};
+
+const allRE = /{[^{}]*}/g;
+const oneRE = /{([^{}]*)}/;
+
+const interpolateOne = (ctx: unknown, str: unknown) => {
+  if (typeof str !== 'string') {
+    return str;
+  }
+
+  // debugging
+  if (str === '{}') {
+    return ctx;
+  }
+
+  const vars = str.match(allRE);
+
+  const processOne = (str: string) => {
+    const match = str.match(oneRE);
+    if (match === null) {
```

Comment:
Can never happen as `allRE` runs first. Ideally one should just use the result from that RE instead of running additional RE that from a statical analysis POV looks like it could fail.

Reply:
I agree, but I opted for not changing any logic when rewriting it in TS. This PR is simply a translation from JS to TS with added tests. No logical changes.